Update SharedPin Pin Map For Scope

deepika-ni opened the pull request:
### What does this Pull Request accomplish?

Updated instruments for Scope as multi-instrument session only includes instrument from the same chassis. 
### Why should this Pull Request be merged?

This PR addresses the changes which is required to validate `SharedPin` support of Scope.

### What testing has been done?

N/A

lornazh commented:
Why do you change the device model? According to the description, you should change only the chassis number?

lornazh commented:
> > Why do you change the device model? According to the description, you should change only the chassis number?
> 
> Based on the instrument availability in this `MixedSignal` offline config file, I have updated this.

I see there is another scope device SCP_5186_C3_S07 in the offline config file, do we want to use this one so that to keep the variety of different models?

deepika-ni commented:
> > > Why do you change the device model? According to the description, you should change only the chassis number?
> > 
> > 
> > Based on the instrument availability in this `MixedSignal` offline config file, I have updated this.
> 
> I see there is another scope device SCP_5186_C3_S07 in the offline config file, do we want to use this one so that to keep the variety of different models?

While trying with this device, I faced this issue, `The driver does not support multi-instrument sessions that use instruments of different models. Ensure that all instruments in the multi-instrument session are identical models.` Is this case possible for Scope?

lornazh commented:
> > > > Why do you change the device model? According to the description, you should change only the chassis number?
> > > 
> > > 
> > > Based on the instrument availability in this `MixedSignal` offline config file, I have updated this.
> > 
> > 
> > I see there is another scope device SCP_5186_C3_S07 in the offline config file, do we want to use this one so that to keep the variety of different models?
> 
> While trying with this device, I faced this issue, `The driver does not support multi-instrument sessions that use instruments of different models. Ensure that all instruments in the multi-instrument session are identical models.` Is this case possible for Scope?

I believe that is a driver requirement when you query for SCP_PIN3, and only for SCP_PIN3. You could still use SCP_5186_C3_S07 and scope device from other chassis for SCP_PIN1 and SCP_PIN2, right?

deepika-ni commented:
> > > > > Why do you change the device model? According to the description, you should change only the chassis number?
> > > > 
> > > > 
> > > > Based on the instrument availability in this `MixedSignal` offline config file, I have updated this.
> > > 
> > > 
> > > I see there is another scope device SCP_5186_C3_S07 in the offline config file, do we want to use this one so that to keep the variety of different models?
> > 
> > 
> > While trying with this device, I faced this issue, `The driver does not support multi-instrument sessions that use instruments of different models. Ensure that all instruments in the multi-instrument session are identical models.` Is this case possible for Scope?
> 
> I believe that is a driver requirement when you query for SCP_PIN3, and only for SCP_PIN3. You could still use SCP_5186_C3_S07 and scope device from other chassis for SCP_PIN1 and SCP_PIN2, right?

I initially assumed so, but when I am trying to retrieve the session for `SCP_PIN1` with the above configuration I am facing this exception in the `Initialize` stage. Kindly correct me If I am missing any steps.

This is the test [method ](https://dev.azure.com/ni/DevCentral/_git/srd-sebu-mixed-signal-apt/pullrequest/982946?path=/SemiconductorTestLibraryDotNET/SemiconductorTestLibraryAbstractions/tests/Unit/InstrumentAbstraction/Scope/InitializeAndCloseTests.cs&version=GBusers/Soliton/user/dkarthik/shared-pin-scope-adjustments&line=48&lineEnd=48&lineStartColumn=14&lineEndColumn=81&type=2&lineStyle=plain&_a=files&iteration=1&base=0)where I faced the exception.

deepika-ni commented:
> > > > > > > Why do you change the device model? According to the description, you should change only the chassis number?
> > > > > > 
> > > > > > 
> > > > > > Based on the instrument availability in this `MixedSignal` offline config file, I have updated this.
> > > > > 
> > > > > 
> > > > > I see there is another scope device SCP_5186_C3_S07 in the offline config file, do we want to use this one so that to keep the variety of different models?
> > > > 
> > > > 
> > > > While trying with this device, I faced this issue, `The driver does not support multi-instrument sessions that use instruments of different models. Ensure that all instruments in the multi-instrument session are identical models.` Is this case possible for Scope?
> > > 
> > > 
> > > I believe that is a driver requirement when you query for SCP_PIN3, and only for SCP_PIN3. You could still use SCP_5186_C3_S07 and scope device from other chassis for SCP_PIN1 and SCP_PIN2, right?
> > 
> > 
> > I initially assumed so, but when I am trying to retrieve the session for `SCP_PIN1` with the above configuration I am facing this exception in the `Initialize` stage. Kindly correct me If I am missing any steps.
> > This is the test [method ](https://dev.azure.com/ni/DevCentral/_git/srd-sebu-mixed-signal-apt/pullrequest/982946?path=/SemiconductorTestLibraryDotNET/SemiconductorTestLibraryAbstractions/tests/Unit/InstrumentAbstraction/Scope/InitializeAndCloseTests.cs&version=GBusers/Soliton/user/dkarthik/shared-pin-scope-adjustments&line=48&lineEnd=48&lineStartColumn=14&lineEndColumn=81&type=2&lineStyle=plain&_a=files&iteration=1&base=0)where I faced the exception.
> 
> I think it's because you're putting both Scope devices in the same "Scope" group. ![image](https://private-user-images.githubusercontent.com/49468811/450564643-f0e61028-1e70-4a8b-807b-f31b5085b193.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg5MzIzMDQsIm5iZiI6MTc0ODkzMjAwNCwicGF0aCI6Ii80OTQ2ODgxMS80NTA1NjQ2NDMtZjBlNjEwMjgtMWU3MC00YThiLTgwN2ItZjMxYjUwODViMTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAzVDA2MjY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlZDVlYmIwZmI4NTM4MDdjNmM0ODU1ODAxZGMyMTVkYjdiNDY4NzFlMzBmZjBjNTJiYjgzNGI2ZjBiYmJkNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kBvAWH_WVQj-TRSjA-G901OP6RzJQsTekyF6ojWE4CE)

Thanks for pointing this, as I removed the group now it is working fine.

lornazh commented on TestAssets/Supporting Materials/Pin Maps/SharedPinTests.pinmap:
```diff
@@ -18,8 +18,8 @@
 		<NIFGenInstrument name="FGEN_5433_C2_S05" numberOfChannels="2" />
-		<NIScopeInstrument name="SCP_5162_C2_S09" numberOfChannels="4" group="Scope" />
-		<NIScopeInstrument name="SCP_5105_C1_S09" numberOfChannels="4" group="Scope" />
+		<NIScopeInstrument name="SCP_5622_C3_S05" numberOfChannels="4" />
```
I guess either change or not change the name is fine.
